randovania/games/am2r/exporter/patch_data_factory.py

```diff
@@ -1,12 +1,227 @@
+            ice_hint = guaranteed_item_hint.create_guaranteed_hints_for_resources(
+                self.description.all_patches,
+                self.players_config,
+                AM2RHintNamer(self.description.all_patches, self.players_config),
+                hint_config.ice_beam == ItemHintMode.HIDE_AREA,
+                ice,
+                False  # TODO: set this to true, when patcher supports setting colors!
+            )
+        else:
+            ice_hint = {k: "Ice Beam is located somewhere on SR-388." for k in ice}
+
+        hints = artifact_hints | ice_hint
+
+        return {
+            key.long_name: value
+            for key, value in hints.items()
+        }
+
+    def _get_item_data(self):
+        item_data = json_lib.read_path(
+            RandovaniaGame.AM2R.data_path.joinpath("pickup_database", "item_data.json")
+        )
+
+        for i in range(1, 47):
+            item_data[f"Metroid DNA {i}"] = item_data["Metroid DNA"]
+
+        item_data["Missiles"] = item_data["Missile Expansion"]
+        item_data["Super Missiles"] = item_data["Super Missile Expansion"]
+        item_data["Power Bombs"] = item_data["Power Bomb Expansion"]
+        item_data["EnergyTransferModule"] = item_data["Nothing"]
+        return item_data
+
     def game_enum(self) -> RandovaniaGame:
         return RandovaniaGame.AM2R
 
     def create_data(self) -> dict:
-        return {}
+        db = self.game
+
+        useless_target = PickupTarget(pickup_creator.create_nothing_pickup(db.resource_database),
+                                      self.players_config.player_index)
+
+        item_data = self._get_item_data()
+        memo_data = {}
+        for (key, value) in item_data.items():
+            memo_data[key] = value["text_desc"]
+        memo_data["Energy Tank"] = memo_data["Energy Tank"].format(Energy=self.patches.configuration.energy_per_tank)
+
+        pickup_list = pickup_exporter.export_all_indices(
+            self.patches,
+            useless_target,
+            self.game.region_list,
+            self.rng,
+            self.configuration.pickup_model_style,
+            self.configuration.pickup_model_data_source,
+            exporter=pickup_exporter.create_pickup_exporter(memo_data, self.players_config, self.game),
+            visual_etm=pickup_creator.create_visual_etm(),
+        )
+
+        return {
+            "configuration_identifier": self._create_hash_dict(),
+            "starting_items": self._create_starting_items_dict(),
+            "starting_location": self._create_starting_location(),
+            "pickups": self._create_pickups_dict(pickup_list, item_data, self.rng),
+            "rooms": self._create_room_dict(),
+            "game_patches": self._create_game_patches(),
+            "door_locks": self._create_door_locks(),
+            "hints": self._create_hints()
+            # TODO: add cosmetic field and decide what to even put in there.
```

Comment:
good candidates for cosmetic settings: default game settings, player appearance, music rando, etc.

Reply:
that being said, you SHOULD add a field and a function for it, even it happens to be an empty dict for the moment

Reply:
Some candidates for cosmetics have been brainstormed. I would add these in an extra PR tho.